docs/script.md

```diff
@@ -320,281 +502,258 @@ In the above snippet the `task.cpus` holds the value for the {ref}`cpus directiv
+`fastq( obj )`
+: Create a FASTQ splitter for an object.
 
-```groovy
-[ 1, 2, 3, 4 ].collect(square)
-```
+`file( filePattern, options = [:] )`
+: Get one or more files from a path or glob pattern. Returns a [Path](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Path.html) or list of Paths if there are multiple files. See [Files and I/O](#files-and-io).
 
-This expression says: Create an array with the values 1, 2, 3 and 4, then call its `collect` method, passing in the closure we defined above. The `collect` method runs through each item in the array, calls the closure on the item, then puts the result in a new array, resulting in:
+`files( filePatten, options = [:] )`
+: Convenience method for `file()` that always returns a list.
 
-```groovy
-[ 1, 4, 9, 16 ]
-```
+`groupKey( key, size )`
+: Create a grouping key to use with the {ref}`operator-grouptuple` operator.
 
-For more methods that you can call with closures as arguments, see the [Groovy GDK documentation](http://docs.groovy-lang.org/latest/html/groovy-jdk/).
+`multiMapCriteria( closure )`
+: Create a multi-map criteria to use with the {ref}`operator-multiMap` operator.
 
-By default, closures take a single parameter called `it`, but you can also create closures with multiple, custom-named parameters. For example, the method `Map.each()` can take a closure with two arguments, to which it binds the `key` and the associated `value` for each key-value pair in the `Map`. Here, we use the obvious variable names `key` and `value` in our closure:
+`sendMail( params )`
+: Send an email. See {ref}`mail-page`.
 
-```groovy
-printMapClosure = { key, value ->
-    println "$key = $value"
-}
+`stop()`
+: Interrupts the iteration when using split operators.
 
-[ "Yue" : "Wu", "Mark" : "Williams", "Sudha" : "Kumari" ].each(printMapClosure)
-```
+`tuple( collection )`
+: Create a tuple object from the given collection.
 
-Prints:
+`tuple( ... args )`
+: Create a tuple object from the given arguments.
 
-```
-Yue = Wu
-Mark = Williams
-Sudha = Kumari
-```
+(implicit-classes)=
 
-A closure has two other important features. First, it can access variables in the scope where it is defined, so that it can interact with them.
+## Implicit classes
 
-Second, a closure can be defined in an anonymous manner, meaning that it is not given a name, and is defined in the place where it needs to be used.
+The following classes are available in Nextflow scripts:
 
-As an example showing both these features, see the following code fragment:
+`Channel`
+: Provides channel factory methods. See {ref}`channel-factory`.
 
-```groovy
-myMap = ["China": 1 , "India" : 2, "USA" : 3]
+`Duration`
+: Represents a duration. See the {ref}`process-time` process directive.
 
-result = 0
-myMap.keySet().each( { result+= myMap[it] } )
+`MemoryUnit`
+: Represents a quantity of bytes. See the {ref}`process-memory` process directive.
```

Comment:
Eventually, would be good to document these classes more thoroughly, and include the unit suffix tables with them, rather than with the time / memory directives. But for now this is fine.

Reply:
Not sure, these were meant to be internal class only

Reply:
Indeed, users should never need to reference `MemoryUnit` directly, but some of its methods, like `toGiga()` and `toMega()`, are useful. Same thing with `Duration`. But I'm not sure yet what is the best way to document that.

Reply:
This issue is relevant: #3567

Reply:
Okay I expanded these sections in a way that makes sense to me. Let me know what you think.